content-review/impl/turnitin-oc/src/main/java/org/sakaiproject/contentreview/turnitin/oc/ContentReviewServiceTurnitinOC.java

```diff
@@ -165,6 +166,77 @@ public void init() {
 		// Populate content upload headers used in uploadExternalContent
 		CONTENT_UPLOAD_HEADERS.putAll(BASE_HEADERS);
 		CONTENT_UPLOAD_HEADERS.put(HEADER_CONTENT, CONTENT_TYPE_BINARY);
+
+		try {
+			String webhookUrl = getWebhookUrl(Optional.empty());
+			if (!webhookUrl.contains("localhost")) {
+				webhookId = setupWebhook(webhookUrl);
+				getWebhook(webhookId);
+			} else {
+				log.info("Running locally - ignoring webhook calls");
+			}
+		} catch (Exception e) {
+			log.error(e.getLocalizedMessage());
+		}
+	}
+
+	public String setupWebhook(String webhookUrl) throws IOException {
+		String id = "";
+
+		Map<String, Object> data = new HashMap<String, Object>();
+
+		List<String> types = new ArrayList<>();
+		types.add("SIMILARITY_COMPLETE");
+		types.add("SUBMISSION_COMPLETE");
+
+		data.put("signing_secret", apiKey);
+		data.put("url", webhookUrl);
+		data.put("description", "test");
+		data.put("allow-insecure", false);
+		data.put("event-types", types);
+
+		HashMap<String, Object> response = makeHttpCall("POST",
+				getNormalizedServiceUrl() + "webhooks",
+				SUBMISSION_REQUEST_HEADERS,
+				data,
+				null);
+
+		// Get response:
+		int responseCode = !response.containsKey(RESPONSE_CODE) ? 0 : (int) response.get(RESPONSE_CODE);
+		String responseMessage = !response.containsKey(RESPONSE_MESSAGE) ? "" : (String) response.get(RESPONSE_MESSAGE);
+		String responseBody = !response.containsKey(RESPONSE_BODY) ? "" : (String) response.get(RESPONSE_BODY);
+
+		if ((responseCode >= 200) && (responseCode < 300) && (responseBody != null)) {
+			// create JSONObject from responseBody
+			JSONObject responseJSON = JSONObject.fromObject(responseBody);
+			if (responseJSON.containsKey("id")) {
+				id = responseJSON.getString("id");
+			} else {
+				throw new Error("Viewer URL not found. Response: " + responseMessage);
+			}
+		} else {
+			throw new Error(responseMessage);
```

Comment:
responseMessage could be null... add more details and variables:
"Invalid request to url: " + url + ", respondeCode: " + respondeCode + ", message: " + responseMessage + ", body: " + responseBody